test/unit/model/RdsResultStorageTest.php

```diff
@@ -83,72 +82,408 @@ public function testStoreRelatedDelivery()
         $this->assertSame($deliveryIdentifier, $this->instance->getDelivery($deliveryResultIdentifier));
     }
 
-    public function testStoreItemVariable()
+    public function testGetAllTestTakerIds()
+    {
+        $deliveryResultIdentifier = "MyDeliveryResultIdentifier#1";
+        $testTakerIdentifier = "mytestTaker#1";
+        $this->instance->storeRelatedTestTaker($deliveryResultIdentifier, $testTakerIdentifier);
+
+        $expected = [
+            [
+                'deliveryResultIdentifier' => $deliveryResultIdentifier,
+                'testTakerIdentifier' => $testTakerIdentifier,
+            ],
+        ];
+
+        $this->assertSame($expected, $this->instance->getAllTestTakerIds());
+    }
+
+    public function testGetAllDeliveryIds()
     {
         $deliveryResultIdentifier = "MyDeliveryResultIdentifier#1";
-        $test = "MyGreatTest#2";
-        $item = "MyGreatItem#2";
-        $callId = "MyCallId#2";
+        $deliveryIdentifier = "mytestTaker#1";
+        $this->instance->storeRelatedDelivery($deliveryResultIdentifier, $deliveryIdentifier);
 
-        $itemVariable = new \taoResultServer_models_classes_OutcomeVariable();
-        $itemVariable->setBaseType('float');
-        $itemVariable->setCardinality('multiple');
-        $itemVariable->setEpoch(microtime());
-        $itemVariable->setIdentifier('Identifier');
-        $itemVariable->setValue('MyValue');
+        $expected = [
+            [
+                'deliveryResultIdentifier' => $deliveryResultIdentifier,
+                'deliveryIdentifier' => $deliveryIdentifier,
+            ],
+        ];
 
+        $this->assertSame($expected, $this->instance->getAllDeliveryIds());
+    }
+
+    public function testCountResultByDelivery()
+    {
+        $deliveryResultIdentifier1 = "MyDeliveryResultIdentifier#1";
+        $deliveryResultIdentifier2 = "MyDeliveryResultIdentifier#2";
+        $testTakerIdentifier1 = "mytestTaker#1";
+        $testTakerIdentifier2 = "mytestTaker#2";
+        $deliveryIdentifier1 = "myDelivery#1";
+        $deliveryIdentifier2 = "myDelivery#2";
+
+        $this->instance->storeRelatedTestTaker($deliveryResultIdentifier1, $testTakerIdentifier1);
+        $this->instance->storeRelatedDelivery($deliveryResultIdentifier1, $deliveryIdentifier1);
+        $this->instance->storeRelatedTestTaker($deliveryResultIdentifier2, $testTakerIdentifier2);
+        $this->instance->storeRelatedDelivery($deliveryResultIdentifier2, $deliveryIdentifier2);
+
+        $this->assertEquals(2, $this->instance->countResultByDelivery([]));
+        $this->assertEquals(1, $this->instance->countResultByDelivery($deliveryIdentifier1));
+    }
+
+    /**
+     * @dataProvider resultByDeliveryToTest
+     *
+     * @param $ids
```

Comment:
if you doc it, maybe you can specify the type ?

Reply:
Right. Done.